### ISSUE TITLE
Multiple groupings are incorrect

### ISSUE THREAD

**Issue by greimel** [JuliaPlots/AlgebraOfGraphics.jl#88]
MWE:

```julia
using DataFrames, AlgebraOfGraphics, CairoMakie, AbstractPlotting

df = DataFrame(grp1 = [0.0, 1.0, 0.0, 1.0],
               grp2 = [0.0, 0.1, 0.1, 0.0])

df = [df; df]

transform!(df, [:grp1, :grp2] => (+) => :y)
df[!,:x] = [0,0,0,0, 1,1,1,1]

data(df) * style(:x, :y, color= :grp1 => categorical, linestyle = :grp2 => categorical) * spec(Lines) |> draw #|> x -> save("/tmp/aog-bug.png", x)
```

![aog-bug](https://user-images.githubusercontent.com/6280307/92031834-84961a00-ed69-11ea-9136-d208bd4cd8d6.png)

The horizontal line at y = 1.0 should be solid.

**Comment by greimel** [comment]
It works after changing the order of arguments:

```julia
using DataFrames, AlgebraOfGraphics, CairoMakie, AbstractPlotting

df = DataFrame(grp1 = [0.0, 1.0, 0.0, 1.0],
                   grp2 = [0.0, 0.1, 0.1, 0.0])

df = [df; df]

transform!(df, [:grp1, :grp2] => (+) => :y)
df[!,:x] = [0,0,0,0, 1,1,1,1]

data(df) * style(:x, :y, linestyle= :grp2 => categorical, color = :grp1 => categorical) * spec(Lines) |> draw #|> x -> save("/tmp/aog-bug.png", x)
```

![aog-good](https://user-images.githubusercontent.com/6280307/92032776-f15de400-ed6a-11ea-81e4-8a488ebed97c.png)

**Comment by piever** [comment]
Closed by #155
